### PR TITLE
abn tutorial

### PR DESCRIPTION
**Pull request by kalantar**
Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>

### CODE REVIEW

File: docs/user-guide/tasks/abnmetrics.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---
+template: main.html
+---
+
+# abnmetrics
+
+Fetch metrics from the Iter8 AB(/n) service.
```
[the remark this answers is not in the document]
Comment (Alan-Cha) [reply docs/user-guide/tasks/abnmetrics.md:7]: Did we go from `A/B(/n)` to `AB(/n)`? I see multiple places where this is true.

**Review by Alan-Cha — CHANGES_REQUESTED** [review]

File: docs/tutorials/abn/abn.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    | kubectl apply -f -
+
+    curl -qs https://raw.githubusercontent.com/kalantar/ab-example/main/backend/deploy.yaml \
+    | sed -e "s#\$NAME#backend#" \
+    | sed -e "s#\$VERSION#v1#" \
+    | sed -e "s#\$TRACK#default#" \
+    | kubectl apply -f -
+    kubectl label deployment backend iter8.tools/abn=true
+    ```
+    3. Generate load.  In separate shells, port-forward requests to the frontend service and generate load for multiple users.  For example:
```
[the remark this answers is not in the document]
Comment (kalantar) [reply docs/tutorials/abn/abn.md]: done

File: docs/tutorials/abn/abn.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+***
+
+## Launch Iter8 A/B(/n) service
+
+If not already deployed, deploy the Iter8 A/B(/n) service. Specify which Kubernetes resource(s) to watch in which namespaces.
+
+```shell
+helm install iter8-abn iter8/iter8-abn \
+--set resources='{deployments,services}' \
+--set namespaces='{default}'     --set image=iter8/iter8:0.11
```
[the remark this answers is not in the document]
Comment (kalantar) [reply docs/tutorials/abn/abn.md]: I removed the `--set image` now that the PR on hub has been merged.

File: docs/tutorials/abn/abn.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+```shell
+kubectl label deployment backend-candidate iter8.tools/abn=true
+```
+
+Once labeled, subsequent requests for a backend will include the candidate version as a valid version to receive traffic.
+
+To terminate traffic to the candidate version, remove the `iter8.tools/abn` label.
+
+??? note "Details about labels"
+    The Iter8 watches resources where the label `iter8-tools/abn` set to `true`. The following labels are expected to be present identifying the role of the resource in an A/B(/n) experiment. Note that an application _version_ might be composed of multiple resources. Iter8 expects only 1 of these resources to be labeled
```
[the remark this answers is not in the document]
Comment (kalantar) [reply docs/tutorials/abn/abn.md]: done

File: docs/user-guide/tasks/abnmetrics.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (kalantar) [reply docs/user-guide/tasks/abnmetrics.md]: fixed.

**Review by Alan-Cha — APPROVED** [review]

**Comment by kalantar** [comment]
Move from v0.11 to v0.12 changed CLI. Should include any updates to commands in other tutorials and CLI documentation.

File: docs/tutorials/custom-metrics/one-version.md [hunk 5]
```diff
@@ -37,7 +37,7 @@ Validate [SLOs](../../getting-started/concepts.md#service-level-objectives) for
 ```shell
 iter8 k launch \
 --set "tasks={custommetrics,assess}" \
---set custommetrics.templates.istio-prom="https://raw.githubusercontent.com/iter8-tools/iter8/v0.12.2-pre/testdata/custommetrics/istio-prom.tpl" \
+--set custommetrics.templates.istio-prom="https://raw.githubusercontent.com/iter8-tools/iter8/v0.12.0/testdata/custommetrics/istio-prom.tpl" \
```
Comment (sriumcp) [inline docs/tutorials/custom-metrics/one-version.md:40]: Fix the location of the template to the "correct" location!

File: docs/tutorials/custom-metrics/two-or-more-versions.md [hunk 6]
```diff
@@ -27,7 +27,7 @@ Validate [SLOs](../../getting-started/concepts.md#service-level-objectives) for
 ```shell
 iter8 k launch \
 --set "tasks={custommetrics,assess}" \
---set custommetrics.templates.istio-prom="https://raw.githubusercontent.com/iter8-tools/iter8/v0.12.2-pre/testdata/custommetrics/istio-prom.tpl" \
+--set custommetrics.templates.istio-prom="https://raw.githubusercontent.com/iter8-tools/iter8/v0.12.0/testdata/custommetrics/istio-prom.tpl" \
```
Comment (sriumcp) [inline docs/tutorials/custom-metrics/two-or-more-versions.md:30]: "Correct" location needed here also.

File: docs/tutorials/abn/abn.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+A/B testing an application's backend component is challenging.
+A/B testing typically relies on business metrics computed by a frontend, user-facing, service.
+Metric values often depend on one or more interactions with backend (not user-facing) components.
+To A/B test a backend component, it is necessary to be able to associate a metric value (computed by the frontend) to the version of the backend component that contributed to its computation.
+The challenge is that the frontend service often does not know which version of the backend component processed a given request.
+This can be achieved, for example, using a service mesh.
+
+To address this challenge, Iter8 introduces an A/B SDK which provides a frontend service with two APIs:
+
+a. **Lookup()** - Identifies a version of a backend service to send a request to. The
```
Comment (sriumcp) [inline docs/tutorials/abn/abn.md:16]: Incomplete sentence?

**Review by sriumcp — CHANGES_REQUESTED** [review]

File: docs/tutorials/custom-metrics/one-version.md (hunk 5 above)
Comment (Alan-Cha) [reply docs/tutorials/custom-metrics/one-version.md:40]: https://github.com/iter8-tools/docs/pull/48

File: docs/tutorials/custom-metrics/two-or-more-versions.md (hunk 6 above)
Comment (Alan-Cha) [reply docs/tutorials/custom-metrics/two-or-more-versions.md:30]: https://github.com/iter8-tools/docs/pull/48

File: docs/tutorials/abn/abn.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
+---
```
Comment (sriumcp) [inline docs/tutorials/abn/abn.md:1]: The excalidraw sources should go under `images/src`.

**Review by sriumcp — CHANGES_REQUESTED** [review]

File: docs/user-guide/tasks/custommetrics.md [hunk 9]
```diff
@@ -13,7 +13,7 @@ In this example, the `custommetrics` task fetches metrics from the Prometheus da
 ```shell
 iter8 k launch \
 --set "tasks={custommetrics,assess}" \
---set custommetrics.templates.istio-prom="https://raw.githubusercontent.com/iter8-tools/iter8/v0.12.2-pre/testdata/custommetrics/istio-prom.tpl" \
+--set custommetrics.templates.istio-prom="https://raw.githubusercontent.com/iter8-tools/iter8/v0.12.0/testdata/custommetrics/istio-prom.tpl" \
```
Comment (sriumcp) [inline docs/user-guide/tasks/custommetrics.md:16]: "Correct" location for template is needed here also.

**Review by sriumcp — CHANGES_REQUESTED** [review]

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**